docs/books/modules/user-guide/adding-metadata-to-connections.adoc

```diff
@@ -0,0 +1,91 @@
+////
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License
+////
+
+// Module is included in the following assemblies:
+//
+// configuring-network-connections.adoc
+
+[id='adding-metadata-to-connections-{context}']
+= Adding metadata to connections
+
+In a complex topology, it can be useful to add metadata to connections so that messages can be handled programmatically.
+
+.Procedure
+
+include::{FragmentDir}/fragment-router-open-config-file-step.adoc[]
+
+. Add arbitrary JSON to the `connector` configuration using the 'openProperties' attribute.
++
+--
+This example adds the property `label` with the value `green`.
+
+[options="nowrap",subs="+quotes"]
+----
+connector {
+   name: broker
+   role: route-container
+   host: 127.0.0.1
+   port: 22180
+   saslMechanisms: ANONYMOUS
+   openProperties: {
+      "label": "green"
+   }
+}
+----
+
+Note the following restrictions on the JSON entries:
+ 
+* ASCII characters only for keys
+* The following keys are not allowed:
+** qd.*
```

Comment:
On re-reading this I'm concerned a reader may take the asterisk literally, like "avoid 'qd.*' but 'qd.foobar' is just fine."

Should probably state explicitly that keys starting with the prefix "qd." and "x-opt-qd." are reserved and should not be used.